### PR TITLE
nasmfmt: unstable-2021-04-24 -> unstable-2022-09-15

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/nasmfmt/default.nix [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
-    rev = "efba220c5252eb717f080d266dcc8304efdeab40";
-    sha256 = "sha256-snhXF+IP0qzl43rKQ0Ugfo1zv3RyNfjxnMpjZEBgPQg=";
+    rev = "127dbe8e72376c67d7dff89010ccfb49fc7b533e";
+    hash = "sha256-1c7ZOdoM0/Us7cnTT3sds2P5pcCedrCfl0GqQBnf9Rk=";
   };
 
-  goPackagePath = "github.com/yamnikov-oleg/nasmfmt";
+  vendorHash = "sha256-pQpattmS9VmO3ZIQUFn66az8GSmB4IvYhTTCFn6SUmo=";
+
+  preBuild = ''
```
Comment (ckiee) [inline pkgs/development/tools/nasmfmt/default.nix:16]: `postPatch`?

**Review by ckiee — COMMENTED** [review]
Builds happily, just a little nit

File: pkgs/development/tools/nasmfmt/default.nix (hunk 1 above)
Comment (aaronjheng) [reply pkgs/development/tools/nasmfmt/default.nix:16]: Unfortunately, not working.

Comment (ckiee) [reply pkgs/development/tools/nasmfmt/default.nix:16]: It builds fine here with `s/preBuild/postPatch/`, x86_64-linux NixOS

Comment (aaronjheng) [reply pkgs/development/tools/nasmfmt/default.nix:16]: If you set vendorHash to empty, you will see:
```
go: go.mod file not found in current directory or any parent directory; see 'go help modules'
```
Per `buildGoModule`, `package` depends on `go-module`. So `go-module` goes first. But there is no `postPatch` hook in `go-module`. Consequently, there is no `go.mod` file when building `go-module`.

**Review by ckiee — APPROVED** [review]

**Pull request merged**